### PR TITLE
bug fixes and improvments when debugging sandbox

### PR DESCRIPTION
**Pull request by n-hutton**
-Make the next aeons file contain two to avoid the scenario where switching keys overwrites an aeon that has generated entropy that is yet to be included in a block (so would be lost on a crash)
-Add some misc metrics and warnings 
-Decrement dkg id by 1 when recovering to avoid bug where it is incremented by dkg runner

### CODE REVIEW

File: beacon/entropy_generator.go [hunk 1]
```diff
@@ -262,16 +264,21 @@ func (entropyGenerator *EntropyGenerator) trimNextAeons() {
...
 				entropyGenerator.nextAeons = entropyGenerator.nextAeons[1:len(entropyGenerator.nextAeons)]
+				changed = true
 			}
 		} else {
-			// Save aeons to file
-			saveAeons(entropyGenerator.baseConfig.NextEntropyKeyFile(), entropyGenerator.nextAeons...)
 			break
 		}
 	}
+
```
Comment (n-hutton) [inline beacon/entropy_generator.go:277]: Prevents next aeons getting written a lot, making it easier to delete it and restart the node

File: beacon/aeon_details.go [hunk 2]
```diff
@@ -244,3 +290,14 @@ func loadAeonDetailsFiles(filePath string) ([]*AeonDetailsFile, error) {
...
+
+// Function to return whether the specified validator is in qual
+func (aeon *aeonDetails) HasValidatorInQual(addr types.Address) bool {
+
+	if aeon.validators == nil {
+		return false
+	}
+
+	return aeon.validators.HasAddress(addr)
+}
```
Comment (jinmannwong) [inline beacon/aeon_details.go:302]: This does not check if the validator is in qual as `aeon.Validators` contains all validators that ran the DKG. You need to use the validator index and check `aeon.aeonExecUnit.InQual(index)`

File: beacon/entropy_generator.go [hunk 3]
```diff
@@ -287,7 +294,8 @@ func (entropyGenerator *EntropyGenerator) resetKeys() bool {
 	if entropyGenerator.aeon != nil && entropyGenerator.lastBlockHeight >= entropyGenerator.aeon.End {
 		// When updating the aeon, we save the current aeon so that in the event of a crash we
 		// can load it since the block height may still be within this old aeon (entropy leads block height)
-		saveAeons(entropyGenerator.baseConfig.OldEntropyKeyFile(), entropyGenerator.aeon)
+		//saveAeons(entropyGenerator.baseConfig.OldEntropyKeyFile(), entropyGenerator.aeon)
+		updateFileAeons(entropyGenerator.baseConfig.OldEntropyKeyFile(), 2, entropyGenerator.aeon)
```
Comment (jinmannwong) [inline beacon/entropy_generator.go:298]: Remove commented out line?

**Review by jinmannwong — CHANGES_REQUESTED** [review]
The `HasValidatorInQual` isn't quite right but otherwise looks good

**Review by jinmannwong — APPROVED** [review]

**Pull request merged**